- Description: cio: Spurious timeouts.
  Symptom:     Not user observable - cio does unneeded I/O reiterations.
  Problem:     After getting an interrupt for killing I/O after a logical vary
               off, the timer is not deleted.
  Solution:    Delete timer when interrupt for killing the I/O arrived.
  Problem-ID:  -
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/cio/device_fsm.c	20 Apr 2004 11:54:31 -0000	1.85.2.1
+++ linux-2.5/drivers/s390/cio/device_fsm.c	23 Apr 2004 12:57:40 -0000	1.85.2.2
@@ -789,6 +789,7 @@
 	struct subchannel *sch;
 
 	sch = to_subchannel(cdev->dev.parent);
+	ccw_device_set_timeout(cdev, 0);
 	/* OK, i/o is dead now. Call interrupt handler. */
 	cdev->private->state = DEV_STATE_ONLINE;
 	if (cdev->handler)

